### PR TITLE
Refactor Selec core component, make it more re-usable

### PR DESCRIPTION
**Pull request by IlyaSavich**
Signed-off-by: Ilya Savich <isavich@box.com>

Refactored `Select` component from `core-components` to make it a bit more customizable.

**What's changed**:

- added `className` prop to add to `FormControl`
- added `showLabel` prop to have ability not to show label (need guys you suggestion here as it's a hack cause I see you use `label` as kinda key, so I can't make it optional without some breaking refactoring)
- added `renderValue` prop to have ability to change value rendering in button
- added `renderOption` prop to have ability to change how options are rendering in the list
- `selected` class is adding to `FormControl` when some value is selected in the field 

**BREAKING CHANGES**

- Property `placeholder` which was used as empty option label and placeholder for input
  will no longer used as empty option label. For this use-case `emptyItemLabel` is added.
- Wrapper `Box` with class `BackstageSelect-root` was removed,
  so now it's possible to have select input width dependent on content
- `native` option will change icon to default browser one
- empty value was changed from `[]` to `''`

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -126,12 +122,17 @@ export type SelectItem = {
 export type SelectedItems = string | string[] | number | number[];
 
 export type SelectProps = {
+  className?: string;
   multiple?: boolean;
   items: SelectItem[];
+  showLabel?: boolean;
```
Comment (IlyaSavich) [inline packages/core-components/src/components/Select/Select.tsx:128]: want to open discussion about this solution here.

It's a hack, I see you use label as kinda key, so I can't make it optional without some breaking refactoring. What can you suggest here ?

**Comment by IlyaSavich** [comment]
@freben can we re-open it ?

**Comment by freben** [comment]
Before I dig into this too far:

There's a constant balancing act of (1) providing a thorough, infinitely adaptable component library, versus (2) providing simplified components whose use is optional, and encouraging people to just use raw material-UI when they have bigger needs.

This PR has good intentions, but the question is, does it start replicating way too much of what material-UI does, and should one just use it directly instead? This is meant as an open question and I am honestly curious how you feel about it :)

**Comment by IlyaSavich** [comment]
Now I face quite often when I need to extend some basic functionality basically I just need to copy this code while I see that just quite small refactoring could fix that issue. That's my original intention

**Pull request closed without merging**